src/Compilers/CSharp/Portable/Syntax/SyntaxTreeDiagnosticEnumerator.cs

```diff
@@ -207,19 +180,11 @@ internal NodeIteration this[int index]
                 }
             }
 
-            internal void UpdateSlotIndexForStackTop(int slotIndex)
-            {
-                Debug.Assert(_stack != null);
-                Debug.Assert(_count > 0);
-                _stack[_count - 1].SlotIndex = slotIndex;
-            }
+            public readonly void UpdateSlotIndexForStackTop(int slotIndex)
+                => _stack[_count - 1].SlotIndex = slotIndex;
```

Comment:
No requirement to do it now, but, in my opinion, the code is going to be clearer with both these functions deleted, and the implementations inlined, and modified to use `Top` instead of `_stack[_count - 1]`.

Reply:
done.